runtime/common/src/xcm.rs

```diff
@@ -158,62 +154,76 @@ where
 /// `MultiLocation`.
 pub struct CurrencyIdConvert<T>(PhantomData<T>);
 
+impl<T> MaybeEquivalence<MultiLocation, CurrencyId> for CurrencyIdConvert<T>
+where
+	T: orml_asset_registry::Config<AssetId = CurrencyId, CustomMetadata = CustomMetadata>
```

Comment:
Where in the XCM logic is this struct used and what does it do?

Reply:
It is required by `staging_xcm_builder::ConvertedConcreteId`. Previously it used `xcm_executor::traits::Convert`

Reply:
But what does it do in the xcm Codebase?

Reply:
It is used to convert between two types back and forth

Reply:
Can you link the code sections in the xcm base? Just am pretty paranoid on the XCM part as we have no real integration tests for that.

Reply:
Sure, the requirement comes from here: https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/xcm-builder/src/asset_conversion.rs#L71